pingcap/tiup
# cluster: copy session certs when scale-out TiDB

### What problem does this PR solve? 

When TiProxy is deployed and then scale out TiDB, the session certs are not copied to the new TiDB instances. Because when judging whether the cluster has TiProxy, the topology is the scale-out spec, not the merged spec.

### What is changed and how it works?

- Split `buildSessionCertTasks` from `buildCertificateTasks`. `buildCertificateTasks` is reverted to that before #2374
- Pass both the `mergedTopo` and `newPart` to `buildSessionCertTasks`. Generate session certs when TiProxy is newly enabled. Copy session certs to both original and new TiDB instances when scale-out TiDB if TiProxy is deployed.

### Check List 

Tests 

 - Unit test
 - No code

Case 1:
1. Deploy a cluster without TiProxy
2. Scale out TiProxy and check the certs and configs of TiDB
3. Scale out TiDB and check the certs and configs of TiDB

Case 2:
1. Deploy a cluster with TiProxy
2. Scale out TiDB and check the certs and configs of TiDB

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation


Release notes:

```release-note
- Fix the bug that after scaling out TiDB, the new TiDB loses signing certs when TiProxy is deployed.
```
